### DISCUSSION TITLE
Unable to customise creation process from within a related table

### DISCUSSION THREAD

**Discussion by declan-bright** [filamentphp/filament#7727] [General]
I may be going about this the wrong way but I'm trying to create comments against a post and so have a comment relationship manager tied to the post but I also need this comment to relate to a user so that I can display who created the comment. So what I'm trying to do on save is get the user id and save this against the new comment. I've tried updating the new relationship manager to follow this from the doc https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-data-before-saving and so looks my code looks like this:

`
class CommentsRelationManager extends RelationManager
{
    protected static string $relationship = 'comments';

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('comment')
                    ->required()
                    ->maxLength(255),
            ]);
    }

    public function table(Table $table): Table
    {
        return $table
            ->recordTitleAttribute('comment')
            ->columns([
                Tables\Columns\TextColumn::make('comment')
                    ->wrap(),
                Tables\Columns\TextColumn::make('user.email')
                ->label('Created By'),
            ])
            ->filters([
                //
            ])
            ->headerActions([
                Tables\Actions\CreateAction::make()
                    ->mutateFormDataUsing(function (array $data): array {
                        $data['user_id'] = auth()->id();
                        return $data;
                    }),
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ])
            ->emptyStateActions([
                Tables\Actions\CreateAction::make(),
            ]);
    }
}
`

However on save I'm getting an error saying the user_id field is not being populated in the database. I've tried this approach on a regular resource action and seems to work okay. Am I missing something on this or is it a potential issue?

**Comment by danharrin** [comment]
Is `user_id` `$fillable`

**Reply by declan-bright** [reply]
Hi Dan, yes just checked and the `user_id` is fillable. I've tried since with dd() statements as well inside the `mutateFormDataUsing` method but doesn't seem to do anything as if the callback function isn't being executed.

**Reply by danharrin** [reply]
Ah I know the problem here, it's because of the empty state action. There are 2 create actions, and the last one overwrites the first one. #7649

**Reply by declan-bright** [reply]
That's sorted it, thanks!
